src/main/java/au/org/ala/biocache/web/DownloadController.java

```diff
@@ -77,147 +91,79 @@ public class DownloadController extends AbstractSecureController {
      * Retrieves all the downloads that are on the queue
      * @return
      */
-    @RequestMapping(value = { "occurrences/offline/download/stats", "occurrences/offline/download/stats.json" }, method = RequestMethod.GET)
-    public @ResponseBody
-    List getCurrentDownloads(
-            HttpServletResponse response,
-            @RequestParam(value = "apiKey", required = true) String apiKey) throws Exception {
-        if (apiKey != null) {
-            if (shouldPerformOperation(apiKey, response, false)) {
-                JsonConfig config = new JsonConfig();
-                config.setJsonPropertyFilter(new PropertyFilter() {
-                    @Override
-                    public boolean apply(Object source, String name, Object value) {
-                        return value == null;
-                    }
-                });
+    @SecurityRequirement(name="JWT")
+    @Secured({"ROLE_ADMIN"})
+    @Operation(summary = "Retrieves all the downloads that are on the queue", tags = "Monitoring")
+    @RequestMapping(value = {"occurrences/offline/download/stats"}, method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public @ResponseBody List getCurrentDownloads() {
+
+        JsonConfig config = new JsonConfig();
+        config.setJsonPropertyFilter((source, name, value) -> value == null);
+
+        JSONArray ja = JSONArray.fromObject(persistentQueueDAO.getAllDownloads(), config);
+        for (Object jo : ja) {
+            String id = (String) ((net.sf.json.JSONObject) jo).get("uniqueId");
+            // TODO  how can we construct these urls
+            ((net.sf.json.JSONObject) jo).put("cancelURL", downloadService.webservicesRoot + "/occurrences/offline/cancel/" + id + "?apiKey=TO_BE_ADDED");
+//            ((net.sf.json.JSONObject) jo).put("cancelURL", downloadService.webservicesRoot + "/occurrences/offline/cancel/" + id + "?apiKey=" + apiKey);
 
-                JSONArray ja = JSONArray.fromObject(persistentQueueDAO.getAllDownloads(), config);
-                for (Object jo : ja) {
-                    String id = (String) ((net.sf.json.JSONObject) jo).get("uniqueId");
-                    ((net.sf.json.JSONObject) jo).put("cancelURL", downloadService.webservicesRoot + "/occurrences/offline/cancel/" + id + "?apiKey=" + apiKey);
-                }
-                return ja;
-            }
         }
-        return null;
+        return ja;
     }
 
     /**
      * Add a download to the offline queue
      * @param requestParams
      * @param ip
-     * @param apiKey
-     * @param type
      * @param response
      * @param request
      * @return
      * @throws Exception
      */
-    @RequestMapping(value = { "occurrences/offline/{type}/download*", "occurrences/offline/{type}/download.json*" } , method = {RequestMethod.GET, RequestMethod.POST})
+    @Operation(summary = "Asynchronous occurrence download", tags = "Download")
```

Comment:
Since this method call authService.getDownloadUser() to get the download user - shouldn't it be marked as a protected API?

Reply:
This endpoint can be configured (`auth.legacy.emailonly.downloads.enabled`) to accept the user's email address as a request parameter. Therefore the endpoint can't me annotated as `@Secured` however it would be advisable to deprecate this behaviour and require clients to authenticate.